test/options_test.cc

```diff
@@ -203,6 +204,67 @@ TEST_F(OptionsImplTest, TlsContext) {
   EXPECT_TRUE(util(*(options_from_proto.toCommandLineOptions()), *cmd));
 }
 
+// We test --multi-target-* options here and not in AlmostAll above because they are mutually
+// exclusive with the URI arg.
+TEST_F(OptionsImplTest, MultiTarget) {
+  Envoy::MessageUtil util;
+  std::unique_ptr<OptionsImpl> options = TestUtility::createOptionsImpl(
+      fmt::format("{} --multi-target-endpoint 1.1.1.1:3 "
+                  "--multi-target-endpoint 2.2.2.2:4 "
+                  "--multi-target-endpoint [::1]:5 "
+                  "--multi-target-endpoint www.example.com:6 "
+                  "--multi-target-path /x/y/z --multi-target-use-https",
+                  client_name_));
+
+  EXPECT_EQ("/x/y/z", options->multiTargetPath());
+  EXPECT_EQ(true, options->multiTargetUseHttps());
+
+  ASSERT_EQ(4, options->multiTargetEndpoints().size());
+  EXPECT_EQ("1.1.1.1", options->multiTargetEndpoints()[0].address().value());
+  EXPECT_EQ(3, options->multiTargetEndpoints()[0].port().value());
+  EXPECT_EQ("2.2.2.2", options->multiTargetEndpoints()[1].address().value());
+  EXPECT_EQ(4, options->multiTargetEndpoints()[1].port().value());
+  EXPECT_EQ("[::1]", options->multiTargetEndpoints()[2].address().value());
+  EXPECT_EQ(5, options->multiTargetEndpoints()[2].port().value());
+  EXPECT_EQ("www.example.com", options->multiTargetEndpoints()[3].address().value());
+  EXPECT_EQ(6, options->multiTargetEndpoints()[3].port().value());
+
+  CommandLineOptionsPtr cmd = options->toCommandLineOptions();
+
+  EXPECT_EQ(cmd->multi_target().use_https().value(), options->multiTargetUseHttps());
+  EXPECT_EQ(cmd->multi_target().path().value(), options->multiTargetPath());
+
+  ASSERT_EQ(4, cmd->multi_target().endpoints_size());
+  EXPECT_EQ(cmd->multi_target().endpoints(0).address().value(), "1.1.1.1");
+  EXPECT_EQ(cmd->multi_target().endpoints(0).port().value(), 3);
+  EXPECT_EQ(cmd->multi_target().endpoints(1).address().value(), "2.2.2.2");
+  EXPECT_EQ(cmd->multi_target().endpoints(1).port().value(), 4);
+  EXPECT_EQ(cmd->multi_target().endpoints(2).address().value(), "[::1]");
+  EXPECT_EQ(cmd->multi_target().endpoints(2).port().value(), 5);
+  EXPECT_EQ(cmd->multi_target().endpoints(3).address().value(), "www.example.com");
+  EXPECT_EQ(cmd->multi_target().endpoints(3).port().value(), 6);
+
+  // Now we construct a new options from the proto we created above. This should result in an
+  // OptionsImpl instance equivalent to options. We test that by converting both to yaml strings,
+  // expecting them to be equal. This should provide helpful output when the test fails by showing
+  // the unexpected (yaml) diff.
+  // The predicates are defined as proto maps, and these seem to re-serialize into a different
+  // order. Hence we trim the maps to contain a single entry so they don't thwart our
+  // textual comparison below.
+  EXPECT_EQ(1, cmd->mutable_failure_predicates()->erase("benchmark.http_4xx"));
```

Comment:
Similar question here - are all of these assertions here related to the new multi-target functionality?

Do we need to keep copying these into all new test cases?

Reply:
The assertion is not particularly relevant, but this "erase" command still needs to be done in order to do the following global proto diff. But we might still keep the assertion as a sanity check since the particular value I remove here is based on defaults that come into play, different from the main `AlmostAll` test.